### PR TITLE
Add support for Maps

### PR DESCRIPTION
**Pull request by Aniket21mathur**
Src -> https://github.com/Aniket21mathur/ProtocolBuffers-Examples/blob/master/chapel/src/maps.proto
Output -> https://github.com/Aniket21mathur/ProtocolBuffers-Examples/blob/master/chapel/build/maps.chpl

### CODE REVIEW

File: src/ProtobufProtocolSupport.chpl [hunk 1]
```diff
@@ -492,6 +492,109 @@ module ProtobufProtocolSupport {
...
+      var (payloadLength, _) = unsignedVarintConsume(ch);
+
+      tagConsume(ch);
+      var key = protoFieldConsumeHelper(ch, protoKeyType, keyType);
+      tagConsume(ch);
+      var value = protoFieldConsumeHelper(ch, protoValueType, valueType);
+      mapField[key] = value;
+    }
+
+    proc protoFieldAppendHelper(val, fieldNumber:int, param protoFieldType, ch:writingChannel) throws {
```
Comment (Aniket21mathur) [inline src/ProtobufProtocolSupport.chpl:522]: @mppf please see this. If this works for us instead of using first class functions.

**Comment by Aniket21mathur** [comment]
This pull request should be rebased with master once #22 is merged.

File: src/ProtobufProtocolSupport.chpl (hunk 1 above)
Comment (mppf) [reply src/ProtobufProtocolSupport.chpl:522]: I like it, anyway :)

**Comment by astatide** [comment]
Everything looks great to me from my end, honestly.  Great work.

**Comment by Aniket21mathur** [comment]
> Everything looks great to me from my end, honestly. Great work.

Thanks for reviewing :)

**Pull request merged**